servers/dsub/jobs/test/base_test_cases.py

```diff
@@ -362,3 +363,58 @@ def test_query_jobs_by_start(self):
                 QueryJobsRequest(start=third_time), [third_job, fourth_job])
             self.assert_query_matches(
                 QueryJobsRequest(start=fourth_time), [fourth_job])
+
+        def test_query_jobs_pagination(self):
```

Comment:
10 jobs might crowd our GCE quota; maybe move this down to the local provider only?

Reply:
Trimmed down to 5 jobs and page size of 2, which I think should still be sufficient to cover the edge cases with less pressure on our quota. Let me know if you think it's still too many.

Reply:
Seems good